compiler/rustc_ty_utils/src/layout.rs

```diff
@@ -673,6 +673,9 @@ fn layout_of_uncached<'tcx>(
 
         // Types with no meaningful known layout.
         ty::Alias(..) => {
+            if ty.has_param() {
```

Comment:
Currently the only problem is that `tests/crashes/114484.rs`, which previously generates an ICE, now it does not reach the recursion limit.